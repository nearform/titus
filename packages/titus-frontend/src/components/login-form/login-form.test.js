import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import LoginForm from '.'

describe('LoginForm', () => {
  afterEach(cleanup)

  it('should not submit if username or password are empty', async done => {
    const mockFn = jest.fn()
    const { findAllByText, getByText } = render(<LoginForm login={mockFn} />)
    const loginButtons = await findAllByText('Login')
    expect(loginButtons.length).toEqual(1)
    fireEvent.click(getByText('Login'))

    setTimeout(() => {
      expect(mockFn).toHaveBeenCalledTimes(1)
      done()
    }, 1)
  })

  it('should render correctly without paramsr', async () => {
    const { asFragment } = render(<LoginForm />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with allowChangePassword and loginError', async () => {
    const { asFragment } = render(
      <LoginForm allowChangePassword={true} loginError="login error" />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
