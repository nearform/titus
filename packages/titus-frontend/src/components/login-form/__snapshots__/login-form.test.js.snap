// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly with allowChangePassword and loginError temporary password 1`] = `
<DocumentFragment>
  <form
    novalidate=""
  >
    <h1 />
    <input
      name="username"
      placeholder="Username"
      required=""
      type="text"
      value=""
    />
    <input
      name="password"
      placeholder="Password"
      required=""
      type="password"
      value=""
    />
    <input
      name="newPassword"
      placeholder="New Password"
      required=""
      type="password"
      value=""
    />
    <div
      class="login__error"
    >
      temporary password
    </div>
    <button
      class="button"
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;

exports[`LoginForm should render correctly without params 1`] = `
<DocumentFragment>
  <form
    novalidate=""
  >
    <h1 />
    <input
      name="username"
      placeholder="Username"
      required=""
      type="text"
      value=""
    />
    <input
      name="password"
      placeholder="Password"
      required=""
      type="password"
      value=""
    />
    <button
      class="button"
      type="submit"
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
